standard/expressions.md

```diff
@@ -2919,6 +2920,24 @@ These are the same transformations applied in [§6.4.3](lexical-structure.md#643
 
 An *anonymous_method_expression* is one of two ways of defining an anonymous function. These are further described in [§11.16](expressions.md#1116-anonymous-function-expressions).
 
+### §throw-expression-operator-new-clause The throw expression operator
```

Comment:
(New problem.)

This is currently within 11.7, so it'll become 11.7.22. The whole of 11.7 is meant to be primary expressions, and this isn't a primary expression, is it? Should we actually list it in 11.14 (the null coalescing operator) as that's the only place it's used?

@BillWagner for initial thoughts; we may need to defer to others. (Sorry, had really hoped I'd be able to merge this.)

Reply:
That would work for me.

@MadsTorgersen @Nigel-Ecma @gafter @ericlippert - any thoughts?

Reply:
@BillWagner, well spotted!

Yes it is in the wrong place, I guess just a "typo" really.

Wherever it goes it does need to stay in the table in 11.4 with the first column updated to include its clause number and the second one to maybe include it by name as well; as already agreed.

As to where it goes I see two options:

Option 1: As Bill suggested add it after 11.14 as a new 11.15

Option 2: make 11.14 into a new "Null Coalescing Expressions", followed by the a grammar showing both the grammar for `null_coalescing_expression` and `throw_expression`, then have 11.14.1 be "The Null Coalescing Operator" followed by the current text of 11.14 sans the grammar, and 11.14.2 be "The Throw Expression" with the new text for throw expressions also sans its grammar. (The grammar section might need to be a "General" sub-clause, so there would be three in total, , I'll defer to @RexJaeschke on that).

Don't pick option 2 if there are plans for throw expressions to be allowed elsewhere in the language in future, otherwise pick whichever seems to fit better. I can see logic in both choices so am ambivalent.

Reply:
Throw expressions will be valid for switch expressions, so I guess a new section is better.

Reply:
> Don't pick option 2 if there are plans for throw expressions to be allowed elsewhere in the language in future, otherwise pick whichever seems to fit better. I can see logic in both choices so am ambivalent.

Coming soon to a standard near you:  `throw` expressions in switch expressions:

```csharp
var result = switch expr
{
  0    => "nil",
  1    => "small",
  <0 => throw new ArgumentExpression("not a good value");
  _   => "more",
}
```

Reply:
I've now moved it to be a new 11.15. @BillWagner PTAL, and we can merge if you're happy.

Reply:
@jskeet Looks like a conflict, and you haven't pushed the latest yet.

Reply:
Hmm, weird. It says it's pushed locally, so I'll need to look. Will do that tomorrow though.

Reply:
> Hmm, weird. It says it's pushed locally, so I'll need to look. Will do that tomorrow though.

I wonder if you pushed to your fork, and not the upstream remote.

Reply:
That was my thought too, but it looks like not. Meetings etc prevent me from sorting it today - will get to it tomorrow.

Reply:
Got it. It was on a different branch in this repo. Have pushed now. PTAL @BillWagner.